src/App.tsx

```diff
@@ -1,34 +1,97 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { SelectTypes } from './types/Enum';
+import { TodoFilter } from './components/TodoFilter';
 
 export const App: React.FC = () => {
+  const [todosList, setTodosList] = useState<Todo[] | null>(null);
+  const [userId, setUserId] = useState(0);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[] | null>(null);
+
+  const getTodosFromApi = useCallback(async () => {
+    const todos = await getTodos().then(todosApi => todosApi);
+
+    if (todos !== undefined) {
+      setTodosList(todos);
+      setVisibleTodos(() => {
+        return todos;
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    getTodosFromApi();
+  }, []);
+
+  const filtered = todosList;
+
+  const search = (query: string, select: SelectTypes) => {
+    const lowercasedQuery = query.toLowerCase();
+
+    if (select === SelectTypes.COMPLETED && todosList) {
+      return todosList.filter(todo => {
+        return todo.title.toLowerCase().includes(lowercasedQuery) && todo.completed;
+      });
+    }
+
+    if (select === SelectTypes.ACTIVE && todosList) {
+      return todosList.filter(todo => todo.title.toLowerCase().includes(lowercasedQuery) && !todo.completed);
+    }
+
+    if (select === SelectTypes.ALL && todosList) {
+      return todosList.filter(todo => todo.title.toLowerCase().includes(lowercasedQuery));
+    }
```

Comment:
if you need to do some logic depending on the value of one variable, will be better to use switch / case instead of if / else